File: C++/Hard/WrongAnswer/handling-sum-queries-after-update_4.cpp
class Solution {
public:
    vector<long long> handleQuery(vector<int>& nums1, vector<int>& nums2, vector<vector<int>>& queries) {
        
        vector<long long> ans;
        vector<long long> sum(nums2.size(), 0);
        for (int i = 0; i < nums2.size(); i++) {
            sum[i] = nums2[i];
            if (i) sum[i] += sum[i - 1];
        }
        vector<long long> diff(nums2.size(), 0);
        for (int i = 0; i < queries.size(); i++) {
            if (queries[i][0] == 1) {
                for (int j = queries[i][1]; j <= queries[i][2]; j++) {
                    nums1[j] ^= 1;
                    diff[j] = -diff[j];
                }
            } else if (queries[i][0] == 2) {
                for (int j = queries[i][1]; j < nums2.size(); j++) {
                    diff[j] += queries[i][2];
                }
            } else {
                ans.push_back(sum[queries[i][2]]);
                if (queries[i][1]) ans.back() -= sum[queries[i][1] - 1];
                for (int j = queries[i][1]; j <= queries[i][2]; j++) {
                    ans.back() += diff[j] * nums1[j];
                }
            }
        }
        return ans;
    }
};

File: Java/Hard/WrongAnswer/handling-sum-queries-after-update_4.java
class Solution {
    public long[] handleQuery(int[] nums1, int[] nums2, int[][] queries) {
        
        int n = nums1.length;
        long[] res = new long[queries.length];
        int[] count = new int[n + 1];
        for (int i = 0; i < n; i++) {
            count[i + 1] = count[i] + nums1[i];
        }
        long sum = 0;
        int index = 0;
        for (int[] query : queries) {
            if (query[0] == 1) {
                int l = query[1], r = query[2];
                sum += (r - l + 1) % 2 * (count[r + 1] - count[l]);
            } else if (query[0] == 2) {
                sum += query[1] * n;
            } else {
                res[index++] = sum;
            }
        }
        return Arrays.copyOf(res, index);
    }
}